src/contexts/AccountContextProvider.tsx

```diff
@@ -46,19 +46,24 @@ export function useUpdateAccountContext() {
 export function AccountContextProvider(props: AccountContextProviderProps): React.ReactElement {
   const { children = null } = props;
   const { keyringPairs, keyringPairsReady } = useKeyringContext();
-  const [account, dispatchAccount] = useReducer(accountReducer, {
+  const [accountState, dispatchAccount] = useReducer(accountReducer, {
     account: null,
-    accounts: []
+    accounts: [],
+    companionAccount: null,
+    senderAccountBalance: null,
+    senderCompanionAccountBalance: null
   });
 
+  useSourceAccountsBalances(accountState, dispatchAccount);
```

Comment:
This is cool, but the only issue I have now is more about directory structure. I imagine that a generic `hooks` folder contains hooks that can be used in any place without any additional dependencies. Now, this particular hook can only be used in conjunction with a reducer and pretty much only in this one place - however by looking into `hooks` directory you won't be able to distinguish "regular hooks" and "helper/specialised hooks" - this long term will imho lead to confusion and "the code is complicated" feeling.

To overcome that we can either put such hooks in a different place, or have the hook `useReducer` internally. The latter is maybe a bit weird, cause if we happen to actually use that hook in other places than the context you will have a separate copy of the state, but well that might actually be a feature really.

Reply:
it's a valid point which makes sense to me. I'm wondering if below the `hooks` directory, i could create a `context` sub folder and place this kind of hooks. Additionally i think it is interesting moving the `useReducer` to this specific dedicated hook.